src/licensedcode/plugin_license.py

```diff
@@ -314,15 +317,15 @@ def add_referenced_filenames_license_matches_for_detections(resource, codebase):
             referenced_license_expression=referenced_license_expression,
             license_detection=license_detection,
         ):
-            if TRACE_REFERENCE:
+            if TRACE_REFERENCE and referenced_resource:
```

Comment:
referenced_resource is declared in another scope, which may be a problem. You may want to track `referenced_resources_with_detections` as a list of Resource rather than `referenced_detections` so you can:
1. debug trace as needed
2. account for when you have more than one file referenced.